### PR TITLE
feat: handles response data type and response as function return type

### PR DESCRIPTION
**Pull request by saad-s**
* defines the symbol for response variable
* sets init value for response variable
* handles none as literal type ( handled as bool )
* fixes function return type

### NOTE: This PR copies response structure handling from #21

### CODE REVIEW

File: src/clarity-frontend/clarity_convert.cpp [hunk 1]
```diff
@@ -4040,6 +4039,135 @@ bool clarity_convertert::get_expr(
...
+ * @brief Creates a response struct symbol based on the given JSON expression and updates the provided type.
+ *
+ * This function is different from create_struct_symbol in that it uses a 
+ * simplistic struct name and id using the expression nodes cid
+ *
+ * @param expr The JSON expression representing the struct.
+ * @param t The type to be updated as a struct type.
+ * @return The created response symbol for the struct.
+ */
+symbolt *clarity_convertert::add_response_symbol_table(
```
Comment (mali-tintash) [inline src/clarity-frontend/clarity_convert.cpp:4052]: should it be "add_response_to_symbol_table" ?

File: src/clarity-frontend/clarity_convert.cpp [hunk 2]
```diff
@@ -4040,6 +4039,135 @@ bool clarity_convertert::get_expr(
...
+    // get the type of second component from leteral type
+    get_type_description(nested_objtype[1], second_comp_type);
+  }
+  else if (response_err_ok_identifier == "err")
+  {
+    name = "err_val";
+    second_comp_name = "ok_val";
+    is_ok = false_exprt();
+    // get the type of second component from leteral type
+    get_type_description(nested_objtype[0], second_comp_type);
```
Comment (mali-tintash) [inline src/clarity-frontend/clarity_convert.cpp:4145]: what value would this attain in case where a response can have both ok and err value ?

**Review by mali-tintash — CHANGES_REQUESTED** [review]

File: src/clarity-frontend/clarity_convert.cpp (hunk 2 above)
Comment (saad-s) [reply src/clarity-frontend/clarity_convert.cpp:4145]: Ok or err whichever is set by response will be set other will be gen_zero

File: src/clarity-frontend/clarity_convert.cpp (hunk 1 above)
Comment (saad-s) [reply src/clarity-frontend/clarity_convert.cpp:4052]: Yes.. will change it

**Review by mali-tintash — APPROVED** [review]

**Pull request merged**